tools/PsesLogAnalyzer/Parse-PsesLog.ps1

```diff
@@ -0,0 +1,268 @@
+                    '^(?<ts>[^\[]+)\[(?<lev>([^\]]+))\]\s+(?<file>..[^:]+):\s+In method\s+''(?<meth>\w+)'',\s+line\s+(?<line>\d+)',
+                    [System.Text.RegularExpressions.RegexOptions]::Compiled -bor [System.Text.RegularExpressions.RegexOptions]::IgnoreCase)
+        }
+        else {
+            # Example new log entry start:
+            # 2018-11-24 12:26:58.302 [DIAGNOSTIC] tid:28 in 'ReadMessage' C:\Users\Keith\GitHub\rkeithhill\PowerShellEditorServices\src\PowerShellEditorServices.Protocol\MessageProtocol\MessageReader.cs: line 114
+            $logEntryRegex =
+                [regex]::new(
+                    '^(?<ts>[^\[]+)\[(?<lev>([^\]]+))\]\s+tid:(?<tid>\d+)\s+in\s+''(?<meth>\w+)''\s+(?<file>..[^:]+):\s+line\s+(?<line>\d+)',
+                    [System.Text.RegularExpressions.RegexOptions]::Compiled -bor [System.Text.RegularExpressions.RegexOptions]::IgnoreCase)
+        }
+
+        $filestream =
+            [System.IO.FileStream]::new(
+                $Path,
+                [System.IO.FileMode]:: Open,
+                [System.IO.FileAccess]::Read,
+                [System.IO.FileShare]::ReadWrite,
+                4096,
+                [System.IO.FileOptions]::SequentialScan)
+
+        $streamReader = [System.IO.StreamReader]::new($filestream, [System.Text.Encoding]::UTF8)
+
+        function nextLine() {
+            if ($null -ne $peekBuf) {
+                $line = $peekBuf
+                $script:peekBuf = $null
+            }
+            else {
+                $line = $streamReader.ReadLine()
+            }
+
+            $script:currentLineNum++
+            $line
+        }
+
+        function peekLine() {
+            if ($null -ne $peekBuf) {
+                $line = $peekBuf;
+            }
+            else {
+                $line = $script:peekBuf = $streamReader.ReadLine()
+            }
+
+            $line
+        }
+
+        function parseLogEntryStart([string]$line) {
+            if ($DebugTimingInfo) {
+                $sw = [System.Diagnostics.Stopwatch]::StartNew()
+            }
+
+            while ($line -notmatch $logEntryRegex) {
+                Write-Warning "Ignoring line:${currentLineNum} '$line'"
+                $line = nextLine
+            }
+
+            if (!$HideProgress -and ($script:logEntryIndex % 50 -eq 0)) {
+                Write-Progress "Processing log entry ${script:logEntryIndex} on line: ${script:currentLineNum}"
+            }
+
+            [string]$timestampStr = $matches["ts"]
+            [DateTime]$timestamp = $timestampStr
+            [PsesLogLevel]$logLevel = $matches["lev"]
+            [int]$threadId = $matches["tid"]
+            [string]$method = $matches["meth"]
+            [string]$file = $matches["file"]
+            [int]$lineNumber = $matches["line"]
+
+            $message = parseLogMessage $method
+
+            [PsesLogEntry]::new($script:logEntryIndex, $timestamp, $timestampStr, $logLevel, $threadId, $method,
```

Comment:
This module targets PS v5.1+ right? Might be worth setting the manifest field. Otherwise, need `New-Object`

Reply:
Better yet, you could use the coercion syntax:

```powershell
[PsesLogEntry]@{
    Index = $script:logEntryIndex
    Timestamp = $timestampStr
    TimestampStr = [datetime]$timestampStr
   ...
}
```

You should even get field completions for that.

Reply:
I set it to 5.1.